sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemTest.java

```diff
@@ -198,6 +199,31 @@ public void queryItems() throws Exception{
         assertThat(feedResponseIterator3.iterator().hasNext()).isTrue();
     }
 
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void queryItemsWithCustomCorrelationActivityId() throws Exception{
+        InternalObjectNode properties = getDocumentDefinition(UUID.randomUUID().toString());
+        CosmosItemResponse<InternalObjectNode> itemResponse = container.createItem(properties);
+
+        String query = String.format("SELECT * from c where c.id = '%s'", properties.getId());
+        CosmosQueryRequestOptions cosmosQueryRequestOptions = new CosmosQueryRequestOptions();
+
+        UUID correlationId = UUID.randomUUID();
+        ImplementationBridgeHelpers
+            .CosmosQueryRequestOptionsHelper
+            .getCosmosQueryRequestOptionsAccessor()
+            .setCorrelationActivityId(cosmosQueryRequestOptions, correlationId);
+
+        CosmosPagedIterable<InternalObjectNode> feedResponseIterator1 =
+            container.queryItems(query, cosmosQueryRequestOptions, InternalObjectNode.class);
+
+        // Very basic validation
+        assertThat(feedResponseIterator1.iterator().hasNext()).isTrue();
+
+        feedResponseIterator1
```

Comment:
Please add validation that the diagnostics for query has the correlated activity id.

Reply:
That has been the case and is validated already. Is also validated in the new end-to-end test I added for the Spark connector.

Reply:
@FabianMeiswinkel  - how would the correlated activity id will be present in query diagnostics?
If it is present, would it be in `queryMetrics` or `FeedResponseDiagnostics`. 
Worth adding a check here, so that in future if someone breaks it, we can catch it here.

Reply:
It is in StoreResult - which is serialized in query diagnostics - not changed by this PR

Reply:
Added an additional assertion just to check that the correlationActivityId is somewhere in getDiagnostcis.tostring

Reply:
Perfect!